[Bug Report] network-agent build fails on master after PR #199 due to stale go.sum

## Summary

The `Build network-agent` job in `Build Check` fails on master with `go: updates to go.mod needed` because PR #199 bumped `CubeNet/cubevs` deps without re-tidying its consumers that pull it in via `replace`.

## Environment

- CubeSandbox version / commit: b2eade2625 (master, run [25713548621](https://github.com/TencentCloud/CubeSandbox/actions/runs/25713548621/job/75498632236))
- Host OS and kernel version: GitHub-hosted `ubuntu-latest` runner (CI environment)
- KVM info (`modinfo kvm`): N/A — failure is at `go build`, before any KVM interaction
- Deployment mode: N/A — CI build failure
- Relevant component: network-agent (build tooling / module graph)

## Steps to Reproduce

1. Check out master at b2eade2625 (or any commit ≥ a661176115 / PR #199).
2. Build the network-agent builder image and run the `network-agent` build target, e.g.:
   ```bash
   make builder-image
   make builder-run      BUILDER_IMAGE="cube-sandbox-builder:ci"      BUILDER_CMD="mkdir -p /workspace/_output/bin &&                   cd /workspace/network-agent &&                   make proto &&                   go build -o /workspace/_output/bin/network-agent ./cmd/network-agent"
   ```
3. Alternatively, with a Go 1.23+ toolchain locally:
   ```bash
   cd network-agent
   go mod tidy -diff   # exits non-zero, listing the missing entries
   ```

## Expected Behavior

`go build ./cmd/network-agent` produces `_output/bin/network-agent` and the CI job succeeds, as it did on the run for commit 6ce4ddba (run [25657770395](https://github.com/TencentCloud/CubeSandbox/actions/runs/25657770395)).

## Actual Behavior

`go build` aborts with the following (last lines of the failed job log):

```
go: downloading google.golang.org/genproto/googleapis/rpc v0.0.0-20251202230838-ff82c1b0f217
...
go: downloading golang.org/x/text v0.32.0
go: updates to go.mod needed; to update it:
        go mod tidy
make: *** [Makefile:85: builder-run] Error 1
##[error]Process completed with exit code 2.
```

### Root cause

PR #199 (a661176115, "CubeVS: Bump go to 1.24.8, golang.org/x/sys to v0.38.0") changed `CubeNet/cubevs/go.mod`:

| Field | Before | After |
|---|---|---|
| `go` directive | 1.22 | 1.24.8 |
| `golang.org/x/sys` | v0.30.0 | v0.38.0 |

`network-agent/go.mod` consumes cubevs via local replace:

```
replace github.com/tencentcloud/CubeSandbox/CubeNet/cubevs => ../CubeNet/cubevs
```

Because of the replace, cubevs's go.mod is part of network-agent's effective module graph. PR #199 only ran `go mod tidy` inside `CubeNet/cubevs/`, so `network-agent/go.mod` and `network-agent/go.sum` are now stale relative to the real import graph, and `go build` refuses to proceed.

### Why this slipped through PR-time CI

- `build-check.yml` filters jobs by changed paths, so PR #199 only ran the `cubevs` job — never the `network-agent` job.
- The last commit to actually touch `network-agent/` (6ce4ddba, IPAM refactor on 2026-05-07) was merged before #199, and at that time cubevs deps were still on the old versions, so its build was clean.
- PR #200 was the first push after #199 that triggered the `network-agent` job, so the failure surfaces against #200's commit despite being unrelated to it.

## Additional Context

### Suggested fix

```bash
cd network-agent
go mod tidy
git add go.mod go.sum
git commit -s -m "chore(network-agent): tidy go.sum after cubevs dep bump in #199"
```

Other modules that `replace` cubevs / cubelog (`Cubelet/`, `CubeMaster/`, `agent/`, …) likely have the same latent skew and should be tidied in the same PR even if their CI jobs have not failed yet.

### Prevention

1. Whenever a shared module like `cubevs` or `cubelog` changes its `go.mod`, run `go mod tidy` in **every consumer that replaces it locally**, in the same PR.
2. Add `go mod tidy -diff` to the `builder-run` step (or as a standalone CI step) so this kind of skew fails fast with a clear message instead of being buried 200 lines into a `go build` log.
3. (Stretch) Consider unifying the multi-module repo with a `go.work` file so intra-repo dep changes propagate to consumers' build graphs automatically.

### Related

- PR #199 — root-cause change: https://github.com/TencentCloud/CubeSandbox/pull/199
- PR #200 — first commit to surface the failure: https://github.com/TencentCloud/CubeSandbox/pull/200
- Failing job: https://github.com/TencentCloud/CubeSandbox/actions/runs/25713548621/job/75498632236
- Last green run on `network-agent`: https://github.com/TencentCloud/CubeSandbox/actions/runs/25657770395

--- a/network-agent/go.mod
+++ b/network-agent/go.mod
@@ -1,6 +1,6 @@
 module github.com/tencentcloud/CubeSandbox/network-agent
 
-go 1.24.0
+go 1.24.8
 
 require (
 	github.com/cilium/ebpf v0.17.3

